libs/data-mapper/src/lib/components/tree/SchemaTreeSearchbar.tsx

```diff
@@ -0,0 +1,182 @@
+    const newFilteredDataTypes: FilteredDataTypesDict = { ...filteredDataTypes };
+
+    if (dataType === selectAll) {
+      Object.keys(filteredDataTypes).forEach((dataTypeKey) => {
+        newFilteredDataTypes[dataTypeKey as DataTypeFilterKeys] = toggledOn;
+      });
+    } else {
+      newFilteredDataTypes[dataType] = toggledOn;
+    }
+
+    setFilteredDataTypes({ ...newFilteredDataTypes });
+  };
+
+  const searchbarInteractions: JSX.Element = (
+    <>
+      {searchTerm && (
+        <Tooltip content={clearLoc} relationship="description">
+          <Button icon={<Dismiss20Regular />} appearance="subtle" onClick={clearSearch} aria-label={clearLoc} size="small" />
+        </Tooltip>
+      )}
+
+      <Menu persistOnItemClick>
+        <MenuTrigger>
+          <Tooltip content={filterLoc} relationship="description">
+            <Button icon={<Filter20Regular />} appearance="subtle" aria-label={filterLoc} size="small" />
+          </Tooltip>
+        </MenuTrigger>
+
+        <MenuPopover>
+          <MenuList>
+            <MenuItem onClick={() => onClickDataType(selectAll, !Object.values(filteredDataTypes).every((isFiltered) => !!isFiltered))}>
+              <Stack horizontal verticalAlign="center">
+                <Checkmark20Filled
+                  style={{
+                    visibility: Object.values(filteredDataTypes).some((isFiltered) => !isFiltered) ? 'hidden' : 'visible',
+                    marginRight: 2,
+                  }}
+                />
+                <Text style={{ color: tokens.colorBrandForegroundLink }}>Select all</Text>
+              </Stack>
+            </MenuItem>
+
+            {Object.entries(NormalizedDataType).flatMap(([dataTypeKey, dataTypeValue]) => {
+              const getMenuItemForDataType = (dataTypeKey: string, dataTypeValue: DataTypeFilterKeys) => {
+                const DataTypeIcon =
+                  dataTypeValue === arrayType
+                    ? iconForNormalizedDataType(NormalizedDataType.ComplexType, 24, true, [SchemaNodeProperty.Repeating])
+                    : iconForNormalizedDataType(dataTypeValue, 24, true);
+
+                return (
+                  <MenuItem key={dataTypeKey} onClick={() => onClickDataType(dataTypeValue, !filteredDataTypes[dataTypeValue])}>
+                    <Stack horizontal verticalAlign="center">
+                      <Checkmark20Filled style={{ visibility: filteredDataTypes[dataTypeValue] ? 'visible' : 'hidden' }} />
+                      <DataTypeIcon
+                        style={{
+                          marginRight: 4,
+                          height: 20,
+                          color: filteredDataTypes[dataTypeValue] ? tokens.colorCompoundBrandForeground1 : undefined,
+                        }}
+                        filled={filteredDataTypes[dataTypeValue]}
+                      />
+                      <Text>{dataTypeKey}</Text>
```

Comment:
We might want to use a localized name or at the very least display name. I think "ComplexType" should at least be "Complex Type" and tbh that might be better as "Object"

Reply:
I agree on 'Object', but otherwise I would say most of the types (the keys of NormalizedDataType which is what's displayed) are already pretty well matched up with what I'd expect to see (Boolean, String, Number, etc)